test/system/220-healthcheck.bats

```diff
@@ -273,4 +273,215 @@ Log[-1].Output   | \"Uh-oh on stdout!\\\nUh-oh on stderr!\\\n\"
     done
 }
 
+@test "podman healthcheck --health-max-log-count default value (5)" {
+    local repeat_count=10
+    local msg="Hello, How are you?"
+    local ctrname="c-h-$(safename)"
+    run_podman run -d --name $ctrname   \
+               --health-cmd "echo $msg" \
+               $IMAGE /home/podman/pause
+    cid="$output"
+
+    run_podman inspect $ctrname --format "{{.Config.HealthMaxLogCount}}"
+    is "$output" "5" "HealthMaxLogCount is set to 5"
+
+    for _ in $(seq 1 $repeat_count);
+    do
+        run_podman healthcheck run $ctrname
+        is "$output" "" "output from 'podman healthcheck run'"
+    done
+
+
+    run_podman inspect $ctrname --format "{{.State.Health.Log}}"
+    count=$(grep -o "$msg" <<< "$output" | wc -l)
+    assert "$count" -eq 5
+
+    run_podman rm -t 0 -f $ctrname
+}
+
+@test "podman healthcheck --health-max-log-count infinite value (0)" {
+    local repeat_count=10
+    local msg="Hello, How are you?"
+    local ctrname="c-h-$(safename)"
+    run_podman run -d --name $ctrname   \
+               --health-cmd "echo $msg" \
+               --health-max-log-count 0 \
+               $IMAGE /home/podman/pause
+    cid="$output"
+
+    run_podman inspect $ctrname --format "{{.Config.HealthMaxLogCount}}"
+    is "$output" "0" "HealthMaxLogCount is set to 0"
+
+    # This is run 11 times to check that the cap is working.
+    for _ in $(seq 0 $repeat_count);
+    do
+        run_podman healthcheck run $ctrname
+        is "$output" "" "output from 'podman healthcheck run'"
+    done
+
+    # sys podman tests executes healthcheck twice the first time healthcheck is run
```

Comment:
This is not correct, you are running them twice... The healthcheck is trigger by podman when the container is started but because you manually run_podman healthcheck run you run it twice not podman.